lib/util/exec/index.ts

```diff
@@ -1,30 +1,20 @@
-import type { ExecOptions as ChildProcessExecOptions } from 'child_process';
 import { dirname, join } from 'upath';
 import { GlobalConfig } from '../../config/global';
 import { TEMPORARY_ERROR } from '../../constants/error-messages';
 import { logger } from '../../logger';
 import { generateInstallCommands } from './buildpack';
-import {
-  DockerExtraCommands,
+import { rawExec } from './common';
+import { generateDockerCommand, removeDockerContainer } from './docker';
+import { getChildProcessEnv } from './env';
+import type {
   DockerOptions,
+  ExecOptions,
   ExecResult,
-  Opt,
+  ExtraEnv,
   RawExecOptions,
-  rawExec,
-} from './common';
-import { generateDockerCommand, removeDockerContainer } from './docker';
-import { getChildProcessEnv } from './env';
-import type { ToolConstraint } from './types';
+} from './types';
 
-type ExtraEnv<T = unknown> = Record<string, T>;
-
-export interface ExecOptions extends ChildProcessExecOptions {
-  cwdFile?: string;
-  extraEnv?: Opt<ExtraEnv>;
-  docker?: Opt<DockerOptions>;
-  toolConstraints?: Opt<ToolConstraint[]>;
-  preCommands?: DockerExtraCommands;
-}
+export * from './types';
```

Comment:
i think we shouldn't do that

Reply:
I saw we did it in a few places, but you think it's not appropriate here? I have no opinion other than that it will just mean a lot of import path changes in other files in this PR.

Reply:
Wasn't a good idea to reexport the types, as we can get cyclic dependencies. and this will cause the `./types.js` to be loaded at runtime. Even if it doesn't contain any values.
```suggestion
```